packages/react-storage/src/components/StorageImage/StorageImage.tsx

```diff
@@ -0,0 +1,36 @@
+import * as React from 'react';
+import classNames from 'classnames';
+
+import { isFunction } from '@aws-amplify/ui';
+import { Image, ComponentClassNames } from '@aws-amplify/ui-react';
+import { useStorageURL } from '@aws-amplify/ui-react/internal';
+
+import type { StorageImageProps } from './types';
+
+export const StorageImage = ({
+  accessLevel,
+  className,
+  fallbackSrc,
+  identityId,
+  imgKey,
+  onStorageGetError,
```

Comment:
Have we considered a simpler name like `onError`?

Reply:
`onError` is the native built-in handler which will trigger when the image failed to load. We need a separate handler to deal with API error. See here - https://github.com/aws-amplify/amplify-ui/pull/4054#discussion_r1231355583